pkg/ibmvpc/sg_analysis_test.go

```diff
@@ -107,3 +110,75 @@ func TestSGRule(t *testing.T) {
+			connections: connection.TCPorUDPConnection(netp.ProtocolString("UDP"), 5, 87, 10, 3245),
+			index:       1,
+			local:       ipblock.GetCidrAll(),
+		},
+		{
+			target:      cidr2t,
+			connections: connection.TCPorUDPConnection(netp.ProtocolString("TCP"), 1, 100, 5, 1000),
+			index:       2,
+			local:       ipblock.GetCidrAll(),
+		},
+	}
+
+	egressConnectivityMap := make(map[*ipblock.IPBlock]*ConnectivityResult)
+	mapAndAnalyzeSGRules(rulesTest1, false, egressConnectivityMap)
+	for _, CR := range egressConnectivityMap {
+		stringCR := CR.string()
+		require.Equal(t, stringCR, "remote: 0.0.0.0-10.250.9.255, conn: No Connections\n"+
+			"remote: 10.250.10.0-10.250.10.0, conn: protocol: TCP src-ports: 1-100 dst-ports: 5-1000\n"+
+			"remote: 10.250.10.1-10.250.10.1, conn: protocol: TCP src-ports: 1-100 dst-ports: 5-1000;"+
+			" protocol: UDP src-ports: 5-87 dst-ports: 10-3245\n"+
+			"remote: 10.250.10.2-10.250.10.3, conn: protocol: TCP src-ports: 1-100 dst-ports: 5-1000\n"+
+			"remote: 10.250.10.4-255.255.255.255, conn: No Connections")
+	}
+}
+
+func TestLocalField(t *testing.T) {
+	cidr1, _ := ipblock.FromCidrOrAddress("10.240.10.1")
+	cidr2, _ := ipblock.FromCidrOrAddress("10.240.10.0/30")
+	cidr1t, _ := ipblock.FromCidrOrAddress("10.250.10.1")
+	cidr2t, _ := ipblock.FromCidrOrAddress("10.250.10.0/30")
+
+	rulesTest1 := []*SGRule{
+		{
+			target:      cidr1t,
+			connections: connection.TCPorUDPConnection(netp.ProtocolString("UDP"), 5, 87, 10, 3245),
+			index:       1,
+			local:       cidr1,
+		},
+		{
+			target:      cidr2t,
+			connections: connection.TCPorUDPConnection(netp.ProtocolString("TCP"), 1, 100, 5, 1000),
+			index:       2,
+			local:       cidr2,
+		},
+	}
+
+	egressConnectivityMap := make(map[*ipblock.IPBlock]*ConnectivityResult)
+	mapAndAnalyzeSGRules(rulesTest1, false, egressConnectivityMap)
+	fullString := ""
+	for _, CR := range egressConnectivityMap {
+		fullString += CR.string()
+	}
+	require.Equal(t, fullString, "remote: 0.0.0.0-10.250.9.255, conn: No Connections\n"+
+		"remote: 10.250.10.0-10.250.10.0, conn: protocol: TCP src-ports: 1-100 dst-ports: 5-1000\n"+
+		"remote: 10.250.10.1-10.250.10.1, conn: protocol: TCP src-ports: 1-100 dst-ports: 5-1000; "+
+		"protocol: UDP src-ports: 5-87 dst-ports: 10-3245\nremote: 10.250.10.2-10.250.10.3, conn: "+
+		"protocol: TCP src-ports: 1-100 dst-ports: 5-1000\nremote: 10.250.10.4-255.255.255.255, "+
+		"conn: No Connectionsremote: 0.0.0.0-10.250.9.255, conn: No Connections\nremote: 10.250.10.0-10.250.10.3,"+
+		" conn: protocol: TCP src-ports: 1-100 dst-ports: 5-1000\nremote: 10.250.10.4-255.255.255.255, conn: No Connections"+
+		"remote: 0.0.0.0-10.250.9.255, conn: No Connections\nremote: 10.250.10.0-10.250.10.3, "+
+		"conn: protocol: TCP src-ports: 1-100 dst-ports: 5-1000\nremote: 10.250.10.4-255.255.255.255, "+
+		"conn: No Connectionsremote: 0.0.0.0-255.255.255.255, conn: No Connectionsremote: "+
+		"0.0.0.0-255.255.255.255, conn: No Connections")
```

Comment:
cannot follow the expected result in the current version..
also, better not use hard-coded strings for tests, as string formatting may change.. 
could be testing using `AllowedConnectivity` results? or through containment check of various connectivity entries in the map?